Victoria & Albert Museum.js

```diff
@@ -0,0 +1,198 @@
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes("/item/")) {
+		return "artwork";
+	}
+	else if (url.includes("/search/")) {
+		return "multiple";
+	}
+	return false;
+}
+
+function scrape(doc, url) {
+	var id = url.match(/item\/(O[0-9]+)/i)[1];
+
+	var apiUrl = "https://api.vam.ac.uk/v2/museumobject/" + id;
+	Zotero.Utilities.doGet(apiUrl, function (text) {
+		var museumobject = JSON.parse(text);
+		var data = museumobject.record;
+		var item = new Zotero.Item("artwork");
+		if (data.titles.length > 0) {
+			item.title = data.titles[0].title;
+			item.shortTitle = data.objectType;
+		}
+		else if (data.objectType) {
+			item.title = data.objectType;
+			item.shortTitle = data.objectType;
+		}
+
+		item.abstractNote = data.briefDescription;
+		item.medium = data.materialsAndTechniques;
+		item.artworkSize = data.dimensionsNote;
+		if (data.productionDates.length > 1) {
+			item.date = data.productionDates[0].date.text + " and further dates";
+		}
+		else if (data.productionDates.length > 0) {
+			item.date = data.productionDates[0].date.text;
+		}
+
+		item.archive = data.collectionCode.text;
+		item.libraryCatalog = 'Victoria & Albert Museum';
+		item.callNumber = data.accessionNumber;
+		item.rights = data.creditLine;
+		item.url = url;
+
+
+		var artistMakerPerson = data.artistMakerPerson;
+		for (var i = 0; i < artistMakerPerson.length; i++) {
+			// This is not ideal and assume surname, firstnames convention is correct
+			var fullName = artistMakerPerson[i].name.text;
+			var lastComma = fullName.lastIndexOf(",");
+			var firstName = "";
+			var lastName = "";
+
+			if (lastComma > 0) {
+				lastName = fullName.substring(0, lastComma);
+				firstName = fullName.substring(lastComma + 1);
+			}
+			else {
+				firstName = fullName;
+			}
+
+			item.creators.push({
+				lastName: lastName,
+				firstName: firstName,
+				creatorType: "author"
+			});
+		}
+
+		var artistMakerPeople = data.artistMakerPeople;
+		for (var j = 0; j < artistMakerPeople.length; j++) {
+			item.creators.push({
+				lastName: artistMakerPeople[j].name.text,
+				fieldMode: 1,
+				creatorType: "author"
+			});
+		}
+
+		var artistMakerOrganisations = data.artistMakerOrganisations;
+		for (var k = 0; k < artistMakerOrganisations.length; k++) {
+			item.creators.push({
+				lastName: artistMakerOrganisations[k].name.text,
+				fieldMode: 1,
+				creatorType: "author"
+			});
+		}
+
+		item.attachments.push({
+			url: url,
```

Comment:
You can just do `document: doc` and omit the `snapshot: true` (because it's implied with `document`). `document` is much faster (and more accurate) than `url` because it can use the already-loaded page.